### ISSUE TITLE
`Story.write_with_links` will ignore everything after the first "page break" in the HTML.

### ISSUE THREAD

**Issue by Julynx** [pymupdf/PyMuPDF#2753]
## Describe the bug

`Story.write` and `Story.write_with_links` will ignore everything after the first `<p style="page-break-after: always;"></p>` in the html.

## To Reproduce

Having the sample HTML content:
```html
<p>Before</p>

<p style="page-break-after: always;"></p>

<p>After</p>
```

1. Create a story from the HTML content
```python
story = fitz.Story(html=html_content,
                   archive=".")
```

2. Write the story to the document and save it to a PDF file 
```python
document = story.write_with_links(rectfn)
document.save(file_path)
document.close()
```

3. The PDF contains only:

Page 1/1
```
Before
```

## Expected behavior
- Ideally, the PDF should contain:

Page 1/2
```
Before
```
Page 2/2
```
After
```

- Or, if the page break functionality is simply not supported, just:

Page 1/1
```
Before
After
```

In no case should it just drop everything before the page break if the HTML happens to have one, which is the observed behavior for me.

## Screenshots

Notice how the PDF just has "Before". There is no "After" text and only one page, besides the HTML having the "After" paragraph after the page break. The code used is the one detailed in "To Reproduce".
![Screenshot](https://github.com/pymupdf/PyMuPDF/assets/64993676/167a9bfa-302d-4847-843a-390cdea2f1c0)

## Your configuration

 - Operating system: Ubuntu 22.04.3 LTS, Kernel 6.2.0-34-generic
 - Python version: Python 3.10.12 [GCC 11.4.0] 
 - PyMuPDF version: PyMuPDF 1.23.5 (Installed through pip)

**Comment by julian-smith-artifex-com** [comment]
Interestingly the `After` text actually appears to be in the PDF (e.g. from `Page.get_text()`) but is not displaying.

**Comment by julian-smith-artifex-com** [comment]
Have created MuPDF bug for this: https://bugs.ghostscript.com/show_bug.cgi?id=707323

**Comment by julian-smith-artifex-com** [comment]
This is now fixed in MuPDF master.

**Comment by julian-smith-artifex-com** [comment]
tests/test_story.py:test_2753() checks this bug is fixed in PyMuPDF-1.23.7 so marking this as fixed in the next release.

**Comment by julian-smith-artifex-com** [comment]
Fixed in 1.23.7.
